PayPalWebPayments/src/main/java/com/paypal/android/paypalwebpayments/PayPalWebCheckoutClient.kt

```diff
@@ -110,14 +128,43 @@ class PayPalWebCheckoutClient internal constructor(
      *
      * @param request [PayPalWebVaultRequest] for vaulting PayPal as a payment method
      */
-    fun vault(
+    suspend fun vault(
         activity: ComponentActivity,
         request: PayPalWebVaultRequest
     ): PayPalPresentAuthChallengeResult {
         vaultSetupTokenId = request.setupTokenId
         analytics.notify(VaultEvent.STARTED, vaultSetupTokenId)
 
-        val result = payPalWebLauncher.launchPayPalWebVault(activity, request)
+        val result = if (request.appSwitchWhenEligible) {
+            // Check app switch eligibility for vault
+            val patchCcoResult = patchCCOWithAppSwitchEligibility(
+                context = activity,
+                orderId = request.setupTokenId,
+                tokenType = TokenType.VAULT_ID,
+                merchantOptInForAppSwitch = request.appSwitchWhenEligible,
+                paypalNativeAppInstalled = true // TODO: implement native app installed check
+            )
+            // Get RedirectUrl
+            val launchUrl = when (patchCcoResult) {
+                is APIResult.Success -> patchCcoResult.data.launchUrl
+                is APIResult.Failure -> null
+            }
+            // if RedirectUrl not available fallback to web vault
+            if (!launchUrl.isNullOrEmpty()) {
+                payPalWebLauncher.launchWithUrl(
+                    activity,
```

Comment:
```suggestion
                    activity.applicationContext,
```

Reply:
Oh wait my bad, I forgot that BrowserSwitch requires an activity reference. My mistake, please disregard.